src/common/hooks/use-diagnostic-permission-prompt.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect } from 'react';
+
+import { ScreenPaths } from '@common/types';
+import { IS_TEST_ENV } from '@common/constants';
+import { userHasAllowedDiagnosticsKey } from '@store/onboarding/onboarding.hooks';
+
+import { useChangeScreen } from './use-change-screen';
+
+export function usePromptUserToSetDiagnosticPermissions() {
+  const changeScreen = useChangeScreen();
+
+  useEffect(() => {
+    const storeValue = localStorage.getItem(userHasAllowedDiagnosticsKey);
+    if (IS_TEST_ENV) return;
+    if (storeValue === null) changeScreen(ScreenPaths.REQUEST_DIAGNOSTICS);
+  }, [changeScreen]);
```

Comment:
```suggestion
  const changeScreen = useChangeScreen();
  const [storeValue] = useHasAllowedDiagnostics()

  useEffect(() => {
    if (IS_TEST_ENV) return;
    if (storeValue === null) changeScreen(ScreenPaths.REQUEST_DIAGNOSTICS);
  }, [changeScreen, storeValue]);
```

Reply:
better if the underlying storage implementation ever changes to something else.

Reply:
tried this approach, but the atomWithStorage doesn't differentiate between unset and default, whereas accessing localstorage directly returns `null` when unset